### PR TITLE
Test: Refactor server/api test to allow part of the tests run in test env

### PR DESCRIPTION
**Pull request by gziolo**
Part of #4915.

This refactoring allows to run at least 2 of the tests in `server/api` test suite in test environment. All other tests require desktop specific oauth secret config values, so we should never expose them to the public. In the long run we should move all that related tests to desktop repository, unless we decide expose that functionality also for Calypso.

/cc @aduth @blowery @mkaz @mjangda

### CODE REVIEW

File: server/api/test/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		localRequest
+			.get( '/logout' )
+			.redirects( 0 )
+			.set( 'cookie', 'wpcom_token=test' )
+			.expect( 'location', config( 'login_url' ) )
+			.expect( 'set-cookie', 'wpcom_token=; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT' )
+			.expect( 302, done );
+	} );
+
+	if ( process.env.NODE_ENV === 'desktop' ) {
```
Comment (gziolo) [inline server/api/test/index.js]: This one is tricky. We need to keep this check unless we mock all desktop specific config settings and whatever else we need to run it in test environment.


@mjangda: Any ideas what we should do about this one? We are using completely independent `test` env for all tests now. We can use test doubles for config or even update config file directly to make it work. Maybe we should even split this task into 2 PRs to make the process more transparent.

**Comment by gziolo** [comment]
@aduth @blowery: can we merge this improvement as it is?

File: server/api/test/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			.get( '/logout' )
+			.redirects( 0 )
+			.set( 'cookie', 'wpcom_token=test' )
+			.expect( 'location', config( 'login_url' ) )
+			.expect( 'set-cookie', 'wpcom_token=; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT' )
+			.expect( 302, done );
+	} );
+
+	// this group works only with desktop oauth setup
+	if ( process.env.NODE_ENV === 'desktop' ) {
```
Comment (aduth) [inline server/api/test/index.js]: An option to consider, in at least showing the tests as skipped:

```
let maybeIt = it;
if ( process.env.NODE_ENV !== 'desktop' ) {
	maybeIt = maybeIt.skip;
}

maybeIt( 'should handle incomplete login details', // ...
```

File: server/api/test/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+describe( 'api', function() {
+	let app, config, localRequest, sandbox;
+
+	allowNetworkAccess();
+	useSandbox( newSandbox => sandbox = newSandbox );
+
+	before( () => {
+		config = require( 'config' );
+		sandbox.stub( config, 'isEnabled' ).returns( true );
```
Comment (aduth) [inline server/api/test/index.js]: Maybe we could be more specific with what's returning `true` here?

```js
sandbox.stub( config, 'isEnabled' ).withArgs( 'oauth' ).returns( true );
```

**Comment by aduth** [comment]
@gziolo I left two minor remarks, but I think it's fine to merge with or without those changes.

File: server/api/test/index.js (hunk 2 above)
Comment (gziolo) [inline server/api/test/index.js]: I like that one, I'll update :)

File: server/api/test/index.js (hunk 3 above)
Comment (gziolo) [inline server/api/test/index.js]: I can add that one, too.

**Pull request merged**